src/services/user/hooks/handle-facebook-create.js

```diff
@@ -0,0 +1,52 @@
+
+const handleFacebookCreate = () => (hook) => {
+  // Pass through this hook if there's no facebook data
+  if (!hook.data.facebookId
+    || !hook.data.facebook
+    || !hook.data.facebook.email
+    || (!hook.data.facebook.name
+      && !hook.data.facebook.first_name
+      && !hook.data.facebook.last_name)
+    ) {
+    return hook;
+  }
+
+  const userService = hook.app.service('/users');
+
+  // Check whether a user with same facebook email exists
+  return userService.find({ query: { email: hook.data.facebook.email } })
+    .then(results => {
+      // Handle answer in both array and object forms
+      const existingUser = results[0] || (results.data && results.data[0]);
+
+      // User properties to be created or updated
+      const data = {
+        facebookId: hook.data.facebookId,
+        email: hook.data.facebook.email,
+        name: hook.data.facebook.name ||
+          `${hook.data.facebook.first_name} ${hook.data.facebook.last_name}`,
+      };
+
+      // Patch existing user's properties
+      if (existingUser) {
+        return userService.patch(existingUser._id, data) // eslint-disable-line no-underscore-dangle
+          .then(updatedUser => {
+            // Set `hook.result` to skip the actual `create` since we updated it already
+            hook.result = updatedUser; // eslint-disable-line no-param-reassign
```

Comment:
I just believe the comment from a github issue. However, after tracking down the source, I'm not sure it's true or not.

The sure thing:
If `hook.result` is not undefined, then, it will skip other before hooks (https://github.com/feathersjs/feathers-hooks/blob/3810913d0fa99be5b91d4229c7d960e0dccadf15/src/hooks.js#L67-L69).

The unsure thing.
After all before hooks, I see the mongoose service would call `create` without checking (https://github.com/feathersjs/feathers-mongoose/blob/a56090cec02481ab6863db80135b52b0c3e3e8fb/src/service.js#L162-L166).